website/src/components/ComponentHeadline.astro

```diff
@@ -2,16 +2,27 @@
 import { fragmentLinkScrollMargin } from '../layouts/base/header/headerConstants';
 interface Props {
     title: string;
+    linkSuffix?: string;
+    collapsible?: boolean;
 }
 
-const { title } = Astro.props;
+const { title, linkSuffix, collapsible } = Astro.props;
 
-const fragmentId = title.toLowerCase().replace(/ /g, '-');
+const fragmentId = title.toLowerCase().replace(/ /g, '-') + (linkSuffix ? `-${linkSuffix}` : '');
 ---
 
-<h2 class={`font-bold ${fragmentLinkScrollMargin} `} id={fragmentId}>
-    <a href={`#${fragmentId}`} class='group flex items-center gap-1 capitalize'>
-        {title}
-        <span class='iconify mdi--link-variant font-normal opacity-0 group-hover:opacity-100'></span>
-    </a>
-</h2>
+{
+    collapsible ? (
+        <a href={`#${fragmentId}`} class='group items-center gap-1 capitalize'>
+            {title}
+            <span class='iconify mdi--link-variant font-normal opacity-0 group-hover:opacity-100' />
+        </a>
+    ) : (
+        <h2 class={`font-bold ${fragmentLinkScrollMargin} `} id={fragmentId}>
+            <a href={`#${fragmentId}`} class='group flex items-center gap-1 capitalize'>
+                {title}
+                <span class='iconify mdi--link-variant font-normal opacity-0 group-hover:opacity-100' />
+            </a>
+        </h2>
+    )
+}
```

Comment:
I think this does the job: The trick was to add `inline` to have it display in the same line as the little triangle of the details summary.
```suggestion
<h2 class={`font-bold capitalize inline ${fragmentLinkScrollMargin}`} id={collapsible ? undefined : fragmentId}>
    {
        collapsible ? (
            title
        ) : (
            <a href={`#${fragmentId}`} class='group flex items-center gap-1'>
                {title}
                <span class='iconify mdi--link-variant font-normal opacity-0 group-hover:opacity-100' />
            </a>
        )
    }
</h2>
```